frame/dapps-staking/src/tests.rs

```diff
@@ -434,3 +438,94 @@ fn withdraw_unbonded_is_not_ok() {
         );
     })
 }
+
+#[test]
+fn register_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        // prepare stash-controller pair with some bonded funds
+        let stash_id = 1;
+        let controller_id = 100;
+        let bond_amount = 200 + EXISTENTIAL_DEPOSIT;
+        let ok_contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+
+        assert_ok!(DappsStaking::bond(
+            Origin::signed(stash_id),
+            controller_id,
+            bond_amount,
+            crate::RewardDestination::Staked
+        ));
+
+        assert_ok!(DappsStaking::register(
+            Origin::signed(controller_id),
+            ok_contract.clone()
+        ));
+
+        System::assert_last_event(mock::Event::DappsStaking(crate::Event::NewContract(
+            ok_contract,
+            bond_amount,
+        )));
+    })
+}
+
+#[test]
+fn register_low_deposit() {
+    ExternalityBuilder::build().execute_with(|| {
+        // prepare stash-controller pair with some bonded funds
+        let stash_id = 1;
+        let controller_id = 100;
+        let bond_amount = 10;
```

Comment:
You can declare a constant in mock and reuse that value here (e.g. REGISTER_DEPOSIT - 1).

Imo, it makes it more readable since it's only important that it's clear that we're bonding less than REGISTER_DEPOSIT value. You can check EXISTENTIAL_DEPOSIT usage in this test file.

Just a suggestion.

Reply:
Done